cli/src/test-utils/test-runner.spec.ts

```diff
@@ -0,0 +1,209 @@
+        params: { id: "def", companyId: "abc" }
+      })
+      .query({ action: "setup" })
+      .reply(200)
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(200);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(true);
+  });
+
+  test("provider state setup fail", async () => {
+    const contract = parseAndCleanse(
+      "./cli/src/test-utils/test-runner-examples/single-provider-state.ts"
+    );
+
+    const scope = nock(baseUrl)
+      .post("/state", { name: "a company exists", params: { id: "abc" } })
+      .query({ action: "setup" })
+      .reply(400)
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(200);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(false);
+  });
+
+  test("provider state teardown fail", async () => {
+    const contract = parseAndCleanse(
+      "./cli/src/test-utils/test-runner-examples/single-provider-state.ts"
+    );
+
+    const scope = nock(baseUrl)
+      .get("/companies/abc")
+      .reply(200, {
+        name: "My Company",
+        employeeCount: 15
+      })
+      .post("/state", { name: "a company exists", params: { id: "abc" } })
+      .query({ action: "setup" })
+      .reply(200)
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(400);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(false);
+  });
+
+  test("response status mismatch", async () => {
+    const contract = parseAndCleanse(
+      "./cli/src/test-utils/test-runner-examples/single-provider-state.ts"
+    );
+
+    const scope = nock(baseUrl)
+      .get("/companies/abc")
+      .reply(204, {
+        name: "My Company",
+        employeeCount: 15
+      })
+      .post("/state", { name: "a company exists", params: { id: "abc" } })
+      .query({ action: "setup" })
+      .reply(200)
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(200);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(false);
+  });
+
+  test.skip("response header mismatch", async () => {
+    const contract = parseAndCleanse(
+      "./cli/src/test-utils/test-runner-examples/no-provider-states.ts"
+    );
+
+    const scope = nock(baseUrl)
+      .post("/companies", { name: "My Company", private: true })
+      .reply(
+        201,
+        { name: "My Company" },
+        { NotLocation: `${baseUrl}/companies/abc` }
+      )
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(200);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(false);
+  });
+
+  test("response body mismatch", async () => {
```

Comment:
Do you already have a test that adds an extra field on top of the expected response? Should it pass? (I know we said it should with Pact, but now that we're rolling out our own testing framework, I wonder if we should question this statement)

Reply:
Good point, I'll add a test for this. When deciding whether or not to pass something for clients - in general I think we should simply ask the question: Will this break my client? If the answer is no, then we allow it